exporter/signalfxexporter/translation/converter.go

```diff
@@ -94,287 +61,294 @@ func NewMetricsConverter(logger *zap.Logger, t *MetricTranslator) *MetricsConver
+
+		out = append(out, &dp)
 	}
-	return dimensions
+	return out
 }
 
-func fromOCMetricDescriptorToMetricType(ocType metricspb.MetricDescriptor_Type) *sfxpb.MetricType {
-	if ocType > metricspb.MetricDescriptor_UNSPECIFIED &&
-		ocType <= metricspb.MetricDescriptor_SUMMARY {
-		return ocDescriptorToMetricType[int(ocType)]
+func makeBaseDataPoint(m pdata.Metric) *sfxpb.DataPoint {
+	return &sfxpb.DataPoint{
+		Metric:     m.Name(),
+		MetricType: fromMetricDataTypeToMetricType(m),
 	}
-
-	return nil
 }
 
-func appendDistributionValues(
-	sfxDataPoints []*sfxpb.DataPoint,
-	sfxBaseDataPoint *sfxpb.DataPoint,
-	distributionValue *metricspb.DistributionValue,
-) []*sfxpb.DataPoint {
-
-	// Translating distribution values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>.
-	sfxDataPoints = appendTotalAndSum(
-		sfxDataPoints,
-		sfxBaseDataPoint,
-		&distributionValue.Count,
-		&distributionValue.Sum)
-
-	// 3. Each histogram bucket is converted to a cumulative counter called
-	// <basename>_bucket and will include a dimension called upper_bound that
-	// specifies the maximum value in that bucket. This metric specifies the
-	// number of events with a value that is less than or equal to the upper
-	// bound.
-	metricName := sfxBaseDataPoint.Metric + "_bucket"
-	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
-	if explicitBuckets == nil {
-		return sfxDataPoints
-	}
-	bounds := explicitBuckets.Bounds
-	sfxBounds := make([]string, len(bounds)+1)
-	for i := 0; i < len(bounds); i++ {
-		sfxBounds[i] = float64ToDimValue(bounds[i])
-	}
-	sfxBounds[len(sfxBounds)-1] = infinityBoundSFxDimValue
+func fromMetricDataTypeToMetricType(metric pdata.Metric) *sfxpb.MetricType {
+	switch metric.DataType() {
 
-	for i, bucket := range distributionValue.Buckets {
+	case pdata.MetricDataTypeIntGauge:
+		return &sfxMetricTypeGauge
 
-		// Adding the "upper_bound" dimension.
-		bucketDimensions := make([]*sfxpb.Dimension, len(sfxBaseDataPoint.Dimensions)+1)
-		copy(bucketDimensions, sfxBaseDataPoint.Dimensions)
+	case pdata.MetricDataTypeDoubleGauge:
+		return &sfxMetricTypeGauge
+
+	case pdata.MetricDataTypeIntSum:
+		if !metric.IntSum().IsMonotonic() {
+			if metric.IntSum().AggregationTemporality() == pdata.AggregationTemporalityDelta {
+				return &sfxMetricTypeCounter
+			}
+			return &sfxMetricTypeGauge
+		}
+		return &sfxMetricTypeCumulativeCounter
 
-		bucketDimensions[len(bucketDimensions)-1] = &sfxpb.Dimension{
-			Key:   upperBoundDimensionKey,
-			Value: sfxBounds[i],
+	case pdata.MetricDataTypeDoubleSum:
+		if !metric.DoubleSum().IsMonotonic() {
+			if metric.DoubleSum().AggregationTemporality() == pdata.AggregationTemporalityDelta {
+				return &sfxMetricTypeCounter
+			}
```

Comment:
We should use counter only if "monotonic && delta" so this if condition should be moved at line 215. Same applies for IntHistogram

```golang
if !metric.DoubleSum().IsMonotonic() {
	return &sfxMetricTypeGauge
}

if metric.IntHistogram().AggregationTemporality() == pdata.AggregationTemporalityDelta {
	return &sfxMetricTypeCounter
}

return &sfxMetricTypeCumulativeCounter
```

Reply:
Histograms don't appear to have any `IsMonotonic` method so should they just always be cumulative counter since they are just accumulating counts of things?

Reply:
For Histogram it depends which field are you talking about:
* Count field is always monotonic -> so will be Counter or CumulativeCounter based on temporality
* Buckets fields are always monotonic -> so will be Counter or CumulativeCounter based on temporality
* Sum I need to think a bit (give me few mins).

Reply:
Based on the proto the Sum cannot be guaranteed to be monotonic, so we should assume it is not monotonic because we can have negative values.

Reply:
So then it can be a non-cumulative counter in Signalfx (those can have negative deltas)?  It seems strange if it is a gauge.